core-android/src/main/java/com/uber/sdk/android/core/auth/LoginActivity.java

```diff
@@ -190,10 +239,21 @@ protected void onNewIntent(Intent intent) {
     }
 
     protected void init() {
-        if(getIntent().getData() != null) {
+        if (getIntent().getData() != null) {
             handleResponse(getIntent().getData());
         } else {
-            loadUrl();
+            sessionConfiguration = (SessionConfiguration) getIntent().getSerializableExtra(EXTRA_SESSION_CONFIGURATION);
```

Comment:
move this into a dedicated and testable function

Reply:
sure